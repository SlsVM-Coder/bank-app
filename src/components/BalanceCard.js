import React from 'react';
import { useSelector } from 'react-redux';

export const BalanceCard = () => {
  const { balance } = useSelector((state) => state.money);
  return (
    <div className="BalanceCard">
      <h2>Balance:</h2>
      {balance && <p className="Number"> {balance}â‚¬</p>}
    </div>
  );
};
