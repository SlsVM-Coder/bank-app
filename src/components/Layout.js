import React from 'react';

export const Layout = ({ component, img, title, children, ...rest }) => {
  return (
    <>
      <div className={`Container ${component}`}>
        <div className="LeftContent">
          <div className="Content">
            {title && <h1> {`${title}`} </h1>}
            {children}
          </div>
        </div>
        <div className="RightContent">
          <img src={require(`./../assets/${img}.png`)} alt={img} />
        </div>
      </div>
    </>
  );
};
