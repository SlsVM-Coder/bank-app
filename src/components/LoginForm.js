import React from 'react';
import { useDispatch } from 'react-redux';
import { useForm } from './../hooks/useForm';
import { startLoginEmailPassword } from './../actions/auth';

export const LoginForm = () => {
  const dispatch = useDispatch();

  const [formValues, handleInputChange] = useForm({
    email: '',
    password: '',
  });

  const { email, password } = formValues;

  const handleLogin = (e) => {
    e.preventDefault();
    dispatch(startLoginEmailPassword(email, password));
  };

  return (
    <form
      className="animate__animated animate__fadeIn animate__faster"
      onSubmit={handleLogin}
    >
      <input
        type="text"
        placeholder="Email"
        name="email"
        className="input-text"
        autoComplete="off"
        value={email}
        onChange={handleInputChange}
      />

      <input
        type="password"
        placeholder="Password"
        name="password"
        className="input-text"
        value={password}
        onChange={handleInputChange}
      />

      <button type="submit" className="btn-primary">
        Sign in
      </button>
    </form>
  );
};
