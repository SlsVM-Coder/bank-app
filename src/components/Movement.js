import React from 'react';

export const Movement = ({
  receiver,
  quantity,
  sender,
  date,
  action,
}) => {
  return (
    <div className="Movement">
      <div className="Details">
        <h3>From: </h3>
        <span>{sender}</span>
        <h3>To: </h3>
        <span>{receiver} </span>
        <h3>Date:</h3>
        <span>{date}</span>
      </div>
      <div className="Quantity">
        <h2>{`${action === 'Transfer' ? '-' : '+'} ${quantity}â‚¬`}</h2>
      </div>
    </div>
  );
};
