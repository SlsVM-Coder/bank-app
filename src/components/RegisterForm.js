import React from 'react';
import { useForm } from './../hooks/useForm';
import { useDispatch, useSelector } from 'react-redux';
import { startRegisterWithEmailPasswordName } from './../actions/auth';
import { setErrorAction, removeError } from './../actions/ui';
import validator from 'validator';

export const RegisterForm = () => {
  const dispatch = useDispatch();
  const { msgError } = useSelector((state) => state.ui);

  const [formValues, handleInputChange] = useForm({
    name: '',
    email: '',
    telephone: '',
    address: '',
    password: '',
    password2: '',
  });
  const { name, email, password, password2, telephone, address } = formValues;

  const handleRegister = (e) => {
    e.preventDefault();

    if (isFormValid()) {
      dispatch(startRegisterWithEmailPasswordName(email, password, name));
    }
  };

  const isFormValid = () => {
    if (name.trim().length === 0) {
      dispatch(setErrorAction('Name is required'));
      return false;
    } else if (!validator.isEmail(email)) {
      dispatch(setErrorAction('Email is not valid'));
      return false;
    } else if (password !== password2 || password.length < 5) {
      dispatch(
        setErrorAction(
          'Password should be at least 6 characters and match each other',
        ),
      );
      return false;
    }
    dispatch(removeError());
    return true;
  };

  return (
    <form
      onSubmit={handleRegister}
      className="animate__animated animate__fadeIn animate__faster"
    >
      {msgError && <div className="auth__alert-error">{msgError}</div>}
      <label htmlFor="name">Full name:</label>
      <input
        type="text"
        placeholder="Full name"
        name="name"
        id="name"
        className="input-text"
        value={name}
        onChange={handleInputChange}
      />
      <label htmlFor="email">Email:</label>
      <input
        type="email"
        placeholder="Email"
        name="email"
        id="email"
        className="input-text"
        value={email}
        onChange={handleInputChange}
      />
      <label htmlFor="telephone">Telephone number:</label>
      <input
        type="tel"
        placeholder="Telephone"
        name="telephone"
        id="telephone"
        className="input-text"
        value={telephone}
        onChange={handleInputChange}
      />
      <label htmlFor="address">Address:</label>
      <input
        type="text"
        placeholder="Address"
        name="address"
        id="address"
        className="input-text"
        value={address}
        onChange={handleInputChange}
      />
      <label htmlFor="password">Password:</label>
      <input
        type="password"
        placeholder="Password"
        name="password"
        id="password"
        className="input-text"
        value={password}
        onChange={handleInputChange}
      />
      <label htmlFor="password2">Repeat password:</label>
      <input
        type="password"
        placeholder="Confirm password"
        name="password2"
        id="password2"
        className="input-text"
        value={password2}
        onChange={handleInputChange}
      />

      <button type="submit" className="btn btn-primary btn-block mb-5">
        Sign up
      </button>
    </form>
  );
};
