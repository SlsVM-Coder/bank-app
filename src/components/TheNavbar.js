import React from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { startLogout } from '../actions/auth';

export const TheNavbar = () => {
  const dispatch = useDispatch();
  const { uid } = useSelector((state) => state.auth);

  const handleSignOut = () => {
    dispatch(startLogout());
  };

  return (
    <nav className="TheNavbar">
      <Link to="/" className="m-2">
        Smart Bank
      </Link>
      {uid ? (
        <button className="btn-primary m-2" onClick={handleSignOut}>
          <span>Sign out</span>
        </button>
      ) : (
        <Link to="/auth">
          <button renderas="button" className="btn-primary m-2">
            <span>Sign in</span>
          </button>
        </Link>
      )}
    </nav>
  );
};
