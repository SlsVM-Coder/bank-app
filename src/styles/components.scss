.btn-primary {
  background: $black;
  padding: 8px 13px;
  color: white;
  font-size: 16px;
  border-radius: $border-radius;
  border: 1px solid transparent;
  cursor: pointer;
  transition: all 0.3s ease;

  &:focus {
    outline: 0;
  }

  &:hover {
    background-color: transparent;
    border: 1px solid $black;
    color: $black;
  }
}

.TheNavbar {
  height: 70px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  box-shadow: 0px 5px 30px rgba(25, 28, 31, 0.08);
  padding: 10px;
}

.input-text {
  color: $dark-grey;
  border: 1px solid $light-grey;
  border-radius: $border-radius;
  padding-left: 10px;
  margin-top: 8px;
  margin-bottom: 15px;
  font-size: 16px;
  height: 30px;
  width: 250px;
  transition: border-bottom 0.3s ease;

  &:focus {
    border: 1px solid $primary-color;
    outline: none;
  }

  &::placeholder {
    font-size: 14px;
  }
}

.BalanceCard {
  height: 170px;
  width: 90%;
  max-width: 350px;
  padding: 10px;
  border: 2px solid $black;
  border-radius: $border-radius;

  .Number {
    margin: 0;
    font-size: 50px;
  }
}

.Movement {
  border-bottom: 3px solid $black;
  padding-bottom: 10px;
  margin-bottom: 10px;
  display: flex;
  align-items: center;

  .Details {
    flex-grow: 1;
    display: grid;
    grid-template-columns: 70px 1fr;
    grid-template-rows: repeat(3, 22px);
    align-items: center;
  }

  .Quantity {
    flex-grow: 1;
    display: flex;
    justify-content: flex-end;
  }
}
