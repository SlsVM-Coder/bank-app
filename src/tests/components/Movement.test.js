import React from 'react';
import { shallow } from 'enzyme';
import { Movement } from '../../components/Movement';

describe('Pruebas en <Movement />', () => {
  test('debe mostrarse correctamente', () => {
    const wrapper = shallow(
      <Movement
        receiver="Test receiver"
        sender="Test sender"
        quantity="40"
        date="Date test"
        action="Deposit"
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
