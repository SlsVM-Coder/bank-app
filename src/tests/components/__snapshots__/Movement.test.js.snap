// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Movement /> debe mostrarse correctamente 1`] = `
<div
  className="Movement"
>
  <div
    className="Details"
  >
    <h3>
      From: 
    </h3>
    <span>
      Test sender
    </span>
    <h3>
      To: 
    </h3>
    <span>
      Test receiver
       
    </span>
    <h3>
      Date:
    </h3>
    <span>
      Date test
    </span>
  </div>
  <div
    className="Quantity"
  >
    <h2>
      + 40â‚¬
    </h2>
  </div>
</div>
`;
